nni/nas/oneshot/pytorch/supermodule/sampling.py

```diff
@@ -76,6 +78,36 @@ def reduction(self, items: list[Any], sampled: list[Any]):
         """Override this to implement customized reduction."""
         return weighted_sum(items)
 
+    def sub_state_dict(self, destination: Any=None, prefix: str='', keep_vars: bool=False):
+        if destination is None:
```

Comment:
Suggest moving the duplicated logics (like `if destination is None: blabla`) into `sub_state_dict` in `BaseSuperNetModule`. And leave a `save_to_sub_state_dict` for subclasses overriding.